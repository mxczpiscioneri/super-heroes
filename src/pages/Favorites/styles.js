import Styled from 'styled-components/native'
import Colors from '../../statics/colors'

export const ListEmpty = Styled.Text`
  background: ${Colors.WHITE};
  color: ${Colors.PRIMARY};
  opacity: 0.9;
  fontSize: 18px;
  textAlign: center;
  padding: 16px;
`
