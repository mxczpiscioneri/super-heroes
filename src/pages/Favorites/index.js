import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { FlatList } from 'react-native'
import FavoriteContext from '../../components/Favorite/context'
import { generateUrlImage } from '../../utils'
import { Background, Character } from '../../components'
import { ListEmpty } from './styles'

class Favorites extends Component {
  static navigationOptions = {
    title: 'Favorites',
  }

  render() {
    const { navigate } = this.props.navigation

    return (
      <Background>
        <FavoriteContext.Consumer>
          {context => (
            <FlatList
              data={Array.isArray(context.favorites) ? context.favorites : []}
              keyExtractor={item => `favorites${item.id}`}
              numColumns={2}
              renderItem={
                ({ item }) => (
                  <Character
                    character={item}
                    legend
                    image={generateUrlImage(item.thumbnail.path, item.thumbnail.extension, 'standard_fantastic')}
                    onPress={() => navigate('Details', { characterId: item.id })}
                  />
                )
              }
              ListEmptyComponent={<ListEmpty>Favorites not found</ListEmpty>}
            />
          )}
        </FavoriteContext.Consumer>
      </Background>
    )
  }
}

Favorites.propTypes = {
  navigation: PropTypes.object.isRequired,
}

export default Favorites
