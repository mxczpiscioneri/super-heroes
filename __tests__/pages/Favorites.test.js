import React from 'react'
import renderer from 'react-test-renderer'
import Favorites from '../../src/pages/Home'
import FavoriteContext from '../../src/components/Favorite/context'

const navigation = {
  navigate: jest.fn(),
  setParams: jest.fn(),
}

describe('Favorites', () => {
  test('Favorites snapshot', () => {
    const rendered = renderer.create(
      <FavoriteContext.Provider>
        <Favorites navigation={navigation} />
      </FavoriteContext.Provider>,
    ).toJSON()
    expect(rendered).toMatchSnapshot()
  })
})
