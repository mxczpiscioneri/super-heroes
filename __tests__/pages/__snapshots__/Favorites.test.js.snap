// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Favorites snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "height": "100%",
        "width": "100%",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/statics/images/background.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "elevation": 2,
        "flex": 1,
        "position": "absolute",
        "shadowOpacity": 0.7,
        "shadowRadius": 5,
        "transform": Array [
          Object {
            "translateY": -80,
          },
        ],
        "zIndex": 10,
      }
    }
  />
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={2}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onEndThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
